src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -46,15 +46,32 @@ def convert_arg(arg: Any) -> Any:
         return arg
 
 
+def iter_size_args(args: tuple[Any, ...]) -> Iterator[int]:
+    for arg in args:
```

Comment:
A short docstring would be nice explaining that we extract size arguments when the domain is implicit and else do not (or a comment in `convert_args`).